File: library_base/src/main/java/cn/flyfun/ktx/gamesdk/base/utils/Logger.kt
package cn.flyfun.ktx.gamesdk.base.utils

import android.os.Handler
import cn.flyfun.support.jarvis.LogRvds

/**
 * @author #Suyghur.
 * Created on 2020/11/30
 */
object Logger {

    private const val TAG: String = "flyfun_game"
    var debug: Boolean = true
    var handler: Handler? = null

    fun i(any: Any) {
        LogRvds.i(TAG, any)
    }

    fun e(msg: String) {
        LogRvds.e(TAG, msg)
    }

    fun d(any: Any) {
        if (debug) {
            LogRvds.d(TAG, any)
        }
    }

    fun logHandler(msg: String) {
        handler?.apply {
            LogRvds.logHandler(this, msg)
        }
    }
}

File: library_core/src/main/cpp/include/logger.cpp
//
// Created by #Suyghur, on 2020/12/18.
//

#include "logger.h"

bool Logger::is_debug = false;

void Logger::init(JNIEnv *env, jobject context) {
    jclass _clz = env->FindClass("cn/flyfun/support/jarvis/OwnDebugUtils");
    if (_clz == NULL) {
        loge("OwnDebugUtils impl clz is NULL !!!");
        return;
    }

    const char *method_name = "isOwnDebug";
    const char *sig = "(Landroid/content/Context;)Z";
    jmethodID jmethod_id = env->GetStaticMethodID(_clz, method_name, sig);
    jboolean jresult = env->CallStaticBooleanMethod(_clz, jmethod_id, context);
    is_debug = jresult == JNI_TRUE;
}


void Logger::logd(const string &msg) {
    if(is_debug){
        LOGD("JNI -> %s", msg.c_str());
    }
}

void Logger::loge(const string &msg) {
    LOGE("JNI -> %s", msg.c_str());
}

void Logger::log_handler(JNIEnv *env, const string &msg) {
    jclass _clz = env->FindClass("cn/flyfun/gamesdk/base/utils/Logger");
    if (_clz == NULL) {
        loge("Logger impl clz is NULL !!!");
        return;
    }
    const char *method_name = "logHandler";
    const char *sig = "(Ljava/lang/String;)V";
    jmethodID jmethod_id = env->GetStaticMethodID(_clz, method_name, sig);
    jstring jmsg = env->NewStringUTF(msg.c_str());
    env->CallStaticVoidMethod(_clz, jmethod_id, jmsg);
}